simpleclient_hibernate/src/main/java/io/prometheus/client/hibernate/HibernateStatisticsCollector.java

```diff
@@ -0,0 +1,446 @@
+package io.prometheus.client.hibernate;
+
+import io.prometheus.client.Collector;
+import io.prometheus.client.CollectorRegistry;
+import io.prometheus.client.CounterMetricFamily;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import org.hibernate.SessionFactory;
+import org.hibernate.stat.Statistics;
+
+/**
+ * Collect metrics from one or more Hibernate SessionFactory instances.
+ * <p>
+ * Usage example:
+ * <pre>
+ * new HibernateStatisticsCollector()
+ *     .add(sessionFactory, "myapp")
+ *     .register();
+ * </pre>
+ * If you are in a JPA environment, you can obtain the SessionFactory like this:
+ * <pre>
+ * SessionFactory sessionFactory =
+ *     entityManager.getEntityManagerFactory().unwrap(SessionFactory.class);
+ * </pre>
+ *
+ * @author Christian Kaltepoth
+ */
+public class HibernateStatisticsCollector extends Collector {
```

Comment:
I think an explicit constrcutor would be clearer to those reading the code, and a convenience constructor that also did the add would be useful too.